extensions/wellinks/actions/checkFlourishCustomer/checkFlourishCustomer.test.ts

```diff
@@ -0,0 +1,137 @@
+import { generateTestPayload } from '../../../../src/tests'
+import {
+  WellinksFlourishClientMockImplementation,
+  WellinksFlourishClient,
+} from '../../__mocks__/wellinksFlourishClient'
+import { mockSettings } from '../../__mocks__/settings'
+import { checkFlourishCustomer } from './checkFlourishCustomer'
+
+jest.mock('../../wellinksFlourishClient', () => ({ WellinksFlourishClient }))
+
+describe('Check Flourish Customer', () => {
+  const onComplete = jest.fn()
+  const onError = jest.fn()
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should call onComplete with a true DataPoint when there is a customer with the given identifier', async () => {
+    const validPayload = generateTestPayload({
+      fields: {
+        identifier: 'identifier',
+      },
+      settings: mockSettings,
+    })
+    WellinksFlourishClientMockImplementation.user.exists.mockImplementationOnce(
+      () => {
+        return true
+      }
+    )
+    await checkFlourishCustomer.onActivityCreated(
+      validPayload,
+      onComplete,
+      onError
+    )
+
+    expect(onError).not.toBeCalled()
+    expect(onComplete).toHaveBeenNthCalledWith(1, {
+      data_points: {
+        userExists: 'true',
+      },
+    })
+  })
+
+  test('should call onComplete with a false DataPoint when there is a customer with the given identifier', async () => {
+    const validPayload = generateTestPayload({
+      fields: {
+        identifier: 'identifier',
+      },
+      settings: mockSettings,
+    })
+    WellinksFlourishClientMockImplementation.user.exists.mockImplementationOnce(
+      () => {
+        return false
+      }
+    )
+    await checkFlourishCustomer.onActivityCreated(
+      validPayload,
+      onComplete,
+      onError
+    )
+
+    expect(onError).not.toBeCalled()
+    expect(onComplete).toHaveBeenNthCalledWith(1, {
+      data_points: {
+        userExists: 'false',
+      },
+    })
+  })
+
+  test('should call onError when the action is not given an identifier', async () => {
+    const validPayload = generateTestPayload({
+      fields: {
+        identifier: undefined,
+      },
+      settings: mockSettings,
+    })
+    WellinksFlourishClientMockImplementation.user.exists.mockImplementationOnce(
+      () => {
+        return true
+      }
+    )
+    await checkFlourishCustomer.onActivityCreated(
+      validPayload,
+      onComplete,
+      onError
+    )
+
+    expect(onComplete).not.toBeCalled()
+    expect(onError).toHaveBeenNthCalledWith(1, {
```

Comment:
given my comment about removing the entire try/catch, you can just make sure the error thrown is a ZodError